### PR TITLE
Implement failed swap handling improvements

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2201,15 +2352,77 @@ impl LightningNode {
...
+                to_address: swap_address_info.address.clone(),
+                sat_per_vbyte,
+            }))
+            .map_to_runtime_error(RuntimeErrorCode::NodeUnavailable, "Coudln't prepare refund")?;
+
+        let send_amount_sats = failed_swap_info.amount.sats - prepare_response.refund_tx_fee_sat;
+
+        if swap_address_info.min_deposit.sats > send_amount_sats {
+            runtime_error!(
+                RuntimeErrorCode::NodeUnavailable,
```
Comment (gcomte) [inline src/lib.rs:2382]: Why that error code? Also for 2 following statements

File: src/lib.rs [hunk 2]
```diff
@@ -2224,72 +2437,19 @@ impl LightningNode {
...
-        // Require onchain fees to be less than half of the onchain balance to leave some leeway
-        //  (for now, the onchain fee is just an estimation because the destination address is unknown)
-        if prepared_sweep.onchain_fee_amount.sats * 2 > onchain_balance.sats_round_down().sats {
-            return Ok(None);
-        }
-
-        let lsp_fees = self.calculate_lsp_fee(prepared_sweep.amount.sats)?;
+        let prepare_onchain_tx =
+            move |to_address: String, sat_per_vbyte: u32| -> Result<(Sats, Sats)> {
+                let x = self
```
Comment (gcomte) [inline src/lib.rs:2442]: maybe `sweep_info` instead of `x`?

File: src/lib.rs [hunk 3]
```diff
@@ -310,16 +311,20 @@ pub struct LightningNode {
...
     /// Only available if enough funds are there to swap.
     pub swap_fees: Option<SwapToLightningFees>,
-    /// Estimate of the fees for sending the funds on-chain using [`LightningNode::sweep`].
-    /// The exact fees will be known when calling [`LightningNode::prepare_sweep`].
+    /// Estimate of the fees for sending the funds on-chain using [`LightningNode::sweep_funds_from_channel_closes`]
+    /// or [`LightningNode::resolve_failed_swap`].
+    /// The exact fees will be known when calling [`LightningNode::prepare_sweep_funds_from_channel_closes`]
+    /// or [`LightningNode::prepare_resolve_failed_swap`].
     pub sweep_onchain_fee_estimate: Amount,
     /// Used internally to sweep with the given on-chain fees.
```
[the remark this answers is not in the document]
Comment (gcomte) [reply src/lib.rs:327]: also it doesn't have to be `pub` I think

[the remark this answers is not in the document]
Comment (danielgranhao) [reply src/lib.rs]: This actually needs to be used by 3L consumers. I kept the old doc comment "User internally..." but it doesn't make much sense. I've reworded the doc comment in [f82d2d9](https://github.com/getlipa/lipa-lightning-lib/pull/1215/commits/f82d2d9d6423d224fcf8aaf9ec5cc80061d4efa3).

File: src/lib.rs [hunk 4]
```diff
@@ -1746,6 +1769,21 @@ impl LightningNode {
         Ok(())
     }
 
+    /// Hides the channel close action required item in case the amount cannot be recovered due
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply src/lib.rs]: Fixed [baf85ff](https://github.com/getlipa/lipa-lightning-lib/pull/1215/commits/baf85ff5b05d98f586259cce4cd1faba0b6d7fdd)

File: src/lib.rs (hunk 2 above)
Comment (danielgranhao) [reply src/lib.rs]: yes, it flew under the radar. Fix [7c74716](https://github.com/getlipa/lipa-lightning-lib/pull/1215/commits/7c74716d858a1fbcd1493fa67c54cd66f1c62385)

File: src/lib.rs (hunk 1 above)
Comment (danielgranhao) [reply src/lib.rs:2385]: I've added some todo comments. The idea is to use the SDK specific error when there is one. Basically get an implementation very similar to `swap_channel_close_funds_to_lightning`.

**Pull request merged**